posthog/api/capture.py

```diff
@@ -534,23 +541,135 @@ def get_event(request):
                             sent_at,
                             event_uuid,
                             token,
-                            extra_headers=[("lib_version", lib_version)],
+                            lib_version,
                         )
                     )
 
                 start_time = time.monotonic()
-                for future in futures:
-                    future.get(timeout=settings.KAFKA_PRODUCE_ACK_TIMEOUT_SECONDS - (time.monotonic() - start_time))
-
+                for future in replay_futures:
+                    if future is not None:
+                        future.get(timeout=settings.KAFKA_PRODUCE_ACK_TIMEOUT_SECONDS - (time.monotonic() - start_time))
+    except ValueError as e:
+        with sentry_sdk.push_scope() as scope:
+            scope.set_tag("capture-pathway", "replay")
+            scope.set_tag("ph-team-token", token)
+            capture_exception(e)
+        # this means we're getting an event we can't process, we shouldn't swallow this
+        # in production this is mostly seen as events with a missing distinct_id
+        return cors_response(
+            request,
+            generate_exception_response("capture", f"Invalid recording payload", code="invalid_payload"),
+        )
+    except KafkaTimeoutError as kte:
```

Comment:
Mostly out of curiosity, are you actually observing these (often?) or this just being defensive?

Reply:
as many as 20 a minute (if I'm reading [this log output correctly](https://grafana.prod-us.posthog.dev/explore?schemaVersion=1&panes=%7B%22d2u%22:%7B%22datasource%22:%22P8E80F9AEF21F6940%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bcontainer%3D%5C%22posthog-recordings%5C%22%7D%20%7C%3D%20%60KafkaTimeoutError%60%20%7C%3D%20%60kafka_session_recording_produce_failure%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22P8E80F9AEF21F6940%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1))

my guess is these are messages that are just way too big for us to process and we're timing out before we're discovering how big they are but 🤷

Reply:
which got me wondering about what we'd do if the data for some session can only time out

i've added a probe for the web sdk's retry_count parameter to try and avoid trying infinitely and DDOSing ourselves 

feels a bit yucky but might protect us

(@marandaneto might be useful if iOS and Android added `retry_count` as a query parameter when sending replays but no rush 😊)